src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -111,17 +121,40 @@ protected override Task DisposeCoreAsync(bool initialized)
                     PropagateCompletion = true
                 };
 
+                // We are taking source blocks from multiple configured projects and creating a SyncLink to combine the sources.
+                // The SyncLink will only publish data when the versions of the sources match. There is a problem with that.
+                // The sources have some version components that will make this impossible to match across TFMs. We introduce a 
+                // intermediate block here that will remove those version components so that the synclink can actually sync versions. 
                 var sourceBlocks = currentProjects.Objects.Select(
-                    cp => cp.Services.ProjectSubscription.JointRuleSource.SourceBlock.SyncLinkOptions<IProjectValueVersions>(sourceLinkOptions));
+                    cp => 
+                    {
+                        var sourceBlock = cp.Services.ProjectSubscription.JointRuleSource.SourceBlock;
+                        var versionDropper = CreateVersionDropperBlock();
+                        _designTimeBuildSubscriptionLinks.Add(sourceBlock.LinkTo(versionDropper, sourceLinkOptions));
```

Comment:
So basically we have zero guarantees that state of the project is in sync here, we still need a long term solution for this.